openadapt/events.py

```diff
@@ -567,31 +610,55 @@ def include_merged_events(to_merge):
 
 
 def discard_unused_events(
-    referred_events, action_events, referred_timestamp_key,
-):
-    referred_event_timestamps = set([
-        getattr(action_event, referred_timestamp_key)
-        for action_event in action_events
-    ])
+    referred_events: list,
+    action_events: list,
+    referred_timestamp_key: str,
+) -> list:
+    """Discard unused events based on the referred timestamp key.
+
+    Args:
+        referred_events (list): The list of referred events.
+        action_events (list): The list of action events.
+        referred_timestamp_key (str): The key representing the timestamp in referred events.
+
+    Returns:
+        list: The filtered list of referred events.
+    """
+    referred_event_timestamps = set(
+        [
+            getattr(action_event, referred_timestamp_key)
+            for action_event in action_events
+        ]
+    )
     num_referred_events_before = len(referred_events)
     referred_events = [
         referred_event
         for referred_event in referred_events
         if referred_event.timestamp in referred_event_timestamps
     ]
     num_referred_events_after = len(referred_events)
-    num_referred_events_removed = (
-        num_referred_events_before - num_referred_events_after
-    )
+    num_referred_events_removed = num_referred_events_before - num_referred_events_after
     logger.info(f"{referred_timestamp_key=} {num_referred_events_removed=}")
     return referred_events
 
 
-def process_events(action_events, window_events, screenshots):
-    # for debugging
-    _action_events = action_events
-    _window_events = window_events
-    _screenshots = screenshots
+def process_events(
+    action_events: list, window_events: list, screenshots: list
+) -> tuple[list, list, list]:
```

Comment:
`tuple[list[ActionEvent], list[WindowEvent], list[Screenshot]]`

Reply:
Addressed in https://github.com/OpenAdaptAI/OpenAdapt/pull/289/commits/d88767ff785961d7eb36ca589d96f707a019b507